### PR TITLE
Support type '/' to search

### PR DESCRIPTION
**Pull request by mu001999**
Related topic on IRLO: https://internals.rust-lang.org/t/rustdoc-use-key-to-search-instead-of-s/20559

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
I remember we had a discussion about changing the key to focus on the search input but can't find it again... I don't really have an opinion on this matter so let's see if everyone else agrees with it.

@rfcbot fcp merge

File: src/librustdoc/html/templates/page.html [hunk 1]
```diff
@@ -135,7 +135,7 @@ <h2>Files</h2> {# #}
                         aria-label="Run search in the documentation" {#+ #}
                         autocomplete="off" {#+ #}
                         spellcheck="false" {#+ #}
-                        placeholder="Click or press ‘S’ to search, ‘?’ for more options…" {#+ #}
+                        placeholder="Type ‘S’ or ‘/’ to search, ‘?’ for more options…" {#+ #}
```
[the remark this answers is not in the document]
Comment (GuillaumeGomez) [reply src/librustdoc/html/templates/page.html:138]: I was thinking about adding the available keys on the right of the search input a bit like what the github search input has.

[the remark this answers is not in the document]
Comment (Manishearth) [reply src/librustdoc/html/templates/page.html:138]: I think maybe just having "/ to search" as a short snippet would be fine. We shouldn't inundiate them with information here, but having "/ to search" on a search box is quite standard and not distracting (github does it).

Having more complex text _is_ distracting

[the remark this answers is not in the document]
Comment (mu001999) [reply src/librustdoc/html/templates/page.html:138]: I agree with just having `'/' to search` or `Type '/' to search`, like what github does

**Comment by wezm** [comment]
Perhaps one thing worth noting is that Firefox has a default binding for `/` that this will potentially override. I don't personally use it myself but I do know it does bother some folks when websites take over browser shortcuts. 

https://support.mozilla.org/en-US/kb/search-contents-current-page-text-or-links?redirectslug=Searching+within+a+page&redirectlocale=en-US#w_quick-find

**Comment by jplatte** [comment]
Actually, the FF thing overrides GitHub's slash-to-search for me, not the other way around (and I would much rather have slash start GitHub's search).

**Comment by wezm** [comment]
> Actually, the FF thing overrides GitHub's slash-to-search for me, not the other way around (and I would much rather have slash start GitHub's search).

That's weird. For me in Firefox (on Linux) `/` jumps to GitHub's search...

**Comment by jplatte** [comment]
🤷🏼 I'm also on [arch] linux, sway, FF dev edition 125.0b9 (though this has been the case for a few releases I'm pretty sure)

**Comment by wezm** [comment]
> 🤷🏼 I'm also on [arch] linux, sway, FF dev edition 125.0b9 (though this has been the case for a few releases I'm pretty sure)

Arch for me as well, although X11 (Awesome) not Wayland. FF 124.0.1

**Comment by timokoesters** [comment]
> 🤷🏼 I'm also on [arch] linux, sway, FF dev edition 125.0b9 (though this has been the case for a few releases I'm pretty sure)

Hi jplatte,  I'm on Arch Linux, sway, FF 124.0.1 too, and for me it is working properly. In Github I can use slash to search and in other tabs slash opens the quickfind menu.

Note that the feature only works when the window is wide enough so that the search bar is visible. Does it work for you on other websites, for example this one should also highlight the search bar when you press slash:
https://duckduckgo.com/?q=test

**Comment by jplatte** [comment]
I think we're going really off-topic here and honestly I don't care that much. You know where to message me in case you want to chat more about this; this GitHub PR doesn't seem like thr right place.

**Comment by GuillaumeGomez** [comment]
With all the new comments since we debated about adding this feature. Should we move forward regardless?

**Comment by Manishearth** [comment]
They don't change my position here imo.

**Pull request merged**